app/src/services/cpk.ts

```diff
@@ -294,15 +288,23 @@ class CPKService {
           operation: CPK.CALL,
           to: collateralAddress,
           value: 0,
-          data: ERC20Service.encodeApproveUnlimited(account),
+          data: ERC20Service.encodeTransfer(account, amount),
         },
-        {
+      ]
+
+      const isAlreadyApprovedForMarketMaker = await conditionalTokens.isApprovedForAll(
```

Comment:
I think it's more readable to move this to the beginning and use a `push`.

Reply:
Sure